icd/api/include/vk_pipeline_cache.h

```diff
@@ -79,7 +69,7 @@ class PipelineCache : public NonDispatchable<VkPipelineCache, PipelineCache>
     VkResult Merge(uint32_t srcCacheCount, const PipelineCache** ppSrcCaches);
 
     VK_INLINE PipelineBinaryCache* GetPipelineCache() const { return m_pBinaryCache; }
-    Vkgc::ICache* GetCacheAdapter() const { return m_pBinaryCache->GetCacheAdapter(); }
+    Vkgc::ICache* GetCacheAdapter() const;
```

Comment:
Why did you move this out of the header file?

Reply:
PipelineBinaryCache is an incomplete type at this point. I'd have to include pipeline_binary_cache.h to make it work. This makes the vk_pipeline_cache header smaller and it's easier to tell what code it depends on.

Reply:
It's only an incomplete type because you removed the include of pipeline_binary_cache.h.  I prefer to keep the include of pipeline_binary_cache.h and keep the function in the header since it is trivial.